cmd/agent/common/import.go

```diff
@@ -144,6 +144,9 @@ func ImportConfig(oldConfigDir string, newConfigDir string, force bool) error {
 		if err := copyFile(src, dst, force); err != nil {
 			return fmt.Errorf("unable to copy %s to %s: %v", src, dst, err)
 		}
+		if err := relocateMinCollectionIntervalInFile(dst); err != nil {
```

Comment:
So this definitely works. I have an idea though that should be easy to implement with the work you've already put in place. Since we're copying the file, we're reading it twice now. I'm wondering if we couldn't just do this once. Maybe we could modify the `copyFile()` function slightly so that it also takes a slice of function types, we could call it `transformations`. And then we would apply those transformations to the `rawData []byte` in sequence. That would be at a high-level...

```golang
type TransformationFunc func(rawData []byte) ([]byte, error)

func copyFile(src, dst string, overwrite boolean, transformations []TransformationFunc) error {
   // do what we currently do, but read src into rawData []byte
   data, err := ioutil.ReadFile(file)
	if err != nil {
		return err
	}

    for _, transformation := range transformations {
        data, err = transformation(data)
    }

    // ...
    return ioutil.WriteFile(file, data, 640)
}
```

Reply:
Thanks for the idea ! Will do.